docker/bin/run-presto

```diff
@@ -20,4 +21,5 @@ if [[ ${NODE_ID_EXISTS} != 0 ]] ; then
     NODE_ID="-Dnode.id=${HOSTNAME}"
 fi
 
-exec /usr/lib/presto/bin/launcher run ${NODE_ID} "$@"
+/usr/lib/presto/bin/launcher run ${NODE_ID} "$@" &
```

Comment:
This means other signals won't get passed to presto, which doesn't seem desirable either.

Reply:
> This means other signals won't get passed to presto, which doesn't seem desirable either.

What other signals would be send by docker or K8s to container?

Reply:
SIGQUIT and, potentially others that presto may have in the future, or any the JVM directly handles for it's own features. SIGUSR1/SIGHUP, etc.

Reply:
It is common in K8s to signal pods explicitly? I'm not even sure if that's possible.
One can always exec into container/pod and signal Presto JVM explicitly.

Reply:
K8s signals pods when a pod begins terminating https://kubernetes.io/docs/concepts/workloads/pods/pod/#termination-of-pods

Reply:
> K8s signals pods when a pod begins terminating

Yes, but it's SIGTERM followed by SIGKILL (when graceful period times-out).

Reply:
Right, Presto could have a signal handler for SIGTERM that calls graceful shutdown. Nothing special about SIGTERM if you setup a signal handler for it.